docs/core/StrategyManager.md

```diff
@@ -0,0 +1,249 @@
+
+**Effects**:
+* `token.safeTransferFrom`: Transfers `amount` of `token` to `strategy` on behalf of the caller.
+* `strategy.deposit`: `StrategyBaseTVLLimits` calculates an exchange rate based on its current token holdings and total share amount. The total share number is then increased, and the number of shares created is returned to the `StrategyManager`. Individual strategies do not track shares on a per-account basis; this is handled in `StrategyManager`.
+* `StrategyManager` awards the Staker with the newly-created shares 
+* `DelegationManager.increaseDelegatedShares`: If the Staker is delegated to an Operator (or is themselves an Operator), the Operator's shares are increased for the strategy in question
+
+**Requirements**:
+* Pause status MUST NOT be set (`StrategyManager`): `PAUSED_DEPOSITS`
+* Caller MUST allow at least `amount` of `token` to be transferred by `StrategyManager`
+* `strategy` in question MUST be whitelisted for deposits. Additionally:
+    * Pause status MUST NOT be set (`StrategyBaseTVLLimits`): `PAUSED_DEPOSITS`
+    * `token` must be the correct `underlyingToken` used by `strategy` (i.e. cannot transfer rETH into the cbETH strategy)
+    * `amount` of `token` transferred must not put `strategy` above its configured deposit caps
+    * `shares` awarded on deposit must be nonzero
+
+*Unimplemented as of M2*:
+* The `onlyNotFrozen` modifier is currently a no-op
+
+#### `depositIntoStrategyWithSignature`
+
+```solidity
+function depositIntoStrategyWithSignature(
+    IStrategy strategy,
+    IERC20 token,
+    uint256 amount,
+    address staker,
+    uint256 expiry,
+    bytes memory signature
+)
+    external
+    onlyWhenNotPaused(PAUSED_DEPOSITS)
+    onlyNotFrozen(staker)
+    nonReentrant
+    returns (uint256 shares)
+```
+
+**Effects**: See `depositIntoStrategy` above. Additionally:
+* The Staker's nonce is incremented
+
+**Requirements**: See `depositIntoStrategy` above. Additionally:
+* Caller MUST provide a valid, unexpired signature over the correct fields
+
+*Unimplemented as of M2*:
+* The `onlyNotFrozen` modifier is currently a no-op
+
+#### `undelegate`
+
+```solidity
+function undelegate() external
+```
+
+Allows a Staker to undelegate from an Operator in the `DelegationManager`, as long as they have no shares in either the `StrategyManager` or `EigenPodManager`. Note that the `StrategyManager` allows undelegation if a Staker is in the withdrawal queue.
+
+**Effects**:
+* `DelegationManager.undelegate`: undelegates the Staker from their delegated Operator
+
+**Requirements**:
+* Staker MUST NOT have shares in any strategy within `StrategyManager`
+* Staker MUST NOT have shares in the `EigenPodManager`
+* Staker MUST NOT be an Operator (in `DelegationManager`)
+
+*Unimplemented as of M2*:
+* The `onlyNotFrozen` modifier is currently a no-op
```

Comment:
this function got move to the DelegationManager. was asked primarily for feedback on the DelegationManager doc, so just noting this here / won't consider it blocking in my review